### PR TITLE
[3576] Preview tweaks

### PR DESCRIPTION
**Pull request by asmega**
### Context

- https://trello.com/c/lD6tJVsR/3576-s-publish-preview-functionality-should-match-find
- There are differences between the Publish preview functionality and how it will actually look on find. This change hopes to remove these differences

### Changes proposed in this pull request

- Links are now prefixed with `http` if there is no protocol specified. This is to prevent incorrect relative urls when they should be external links
- Smart quotes used when markdown is processed
- Markdown processor does some additional work to better render lists and bullet points to due incorrect markdown entered by users
-  Copy changes to the preview pages so it matches Find
- When a course is in `rolled_over` state and is being previewed we now show the "Apply section" as part of the preview

### Guidance to review

- Find a course in rolled over state and preview
- Compare with current version on Find
- These 2 pages should be near to identical

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Product Review

### CODE REVIEW

**Comment by asmega** [comment]
[[S] Publish preview functionality should match Find](https://trello.com/c/lD6tJVsR/3576-s-publish-preview-functionality-should-match-find)

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -13,6 +13,21 @@ def markdown(source)
...
+
+    # Fix common markdown errors:
+    # - using bullets rather than *
+    # - not putting a space between * and word
+    source = source.gsub(/•\s?/, "* ").gsub(/^\*(?![\s\*])/, "* ")
+
+    # Convert quotes to smart quotes
+    source_with_smart_quotes = smart_quotes(source)
+    markdown.render(source_with_smart_quotes).html_safe
+  end
```
Comment (fofr) [inline app/helpers/application_helper.rb:25]: I'd purposefully omitted this from Publish preview, as I'd prefer providers to put in the correct markdown, and then fix it if it's broken when checking on preview.

Comment (asmega) [reply app/helpers/application_helper.rb:25]: i see your point but at the same time don't users want to see the actual preview? I guess it's a double edged sword. I think i will leave smart quotes in and remove bullet point one

Comment (asmega) [reply app/helpers/application_helper.rb]: line 20 has now gone

**Pull request merged**